app/controllers/api/note_subscriptions_controller.rb

```diff
@@ -0,0 +1,22 @@
+module Api
+  class NoteSubscriptionsController < ApiController
+    before_action :check_api_writable
+    before_action :authorize
+
+    authorize_resource
+
+    def create
+      note_id = params[:note_id].to_i
+      note = Note.find(note_id)
+      note.subscribers << current_user
+    rescue ActiveRecord::RecordNotUnique
+      head :conflict
```

Comment:
Is `conflict` right here? This happens if you try and subscribe to a note you're already subscribed to right?

Reply:
`conflict` is what changeset subscribe does in this case, I'm doing the same here

https://wiki.openstreetmap.org/wiki/API_v0.6#Subscribe:_POST_/api/0.6/changeset/#id/subscribe

Reply:
OK but that does raise a custom error that includes a nice message explaining the problem - should we do that here? and for the not found case on unsubscription.

I'm not really sure why changesets do it by throwing an exception - a custom message could be just as easily returned inline.

Reply:
added error messages